develop/blocks/basics.md

```diff
@@ -0,0 +1,113 @@
+
+This tutorial explains the basics: a cubic block with no associated block entity.
+
+## Simple blocks
+Let's add a block with no functionality, named "border". To do so, first create a `Block` instance and store it in a `static final` field in your class (usually your `ModInitializer`):
+
+@[code lang=java transcludeWith=:::1](@/reference/latest/src/main/java/com/example/docs/block/FabricDocsReferenceBlocks.java)
+
+The `Block` constructor takes an `AbstractBlock.Settings` instance. This configures the properties common to all block states. Here, `strength(1.5f)` specifies the block's hardness; this is the same as that of natural stones. (Note that this does **not** specify that you need a pickaxe to mine this block.) There are many settings, from sounds to light level and map colors; try experiment with them!
+
+Then, inside your `ModInitializer`'s `onInitialize` method, the block instance must be registered in `Registries.BLOCK`, like this:
+
+@[code lang=java transcludeWith=:::2](@/reference/latest/src/main/java/com/example/docs/block/FabricDocsReferenceBlocks.java)
+
+Note that the item for your block is registered separately. Some blocks (like fire) do not have an item form, but most do. To register the corresponding item, create a `BlockItem` instance and register it under `Registries.ITEM`.
+
+@[code lang=java transcludeWith=:::3](@/reference/latest/src/main/java/com/example/docs/block/FabricDocsReferenceBlocks.java)
+
+## Block models and textures
+A block has a model, which describes the shape and textures of the rendered block. (This is different from the shape used for checking collision.) A block can have multiple models depending on the block state; the model is picked according to the "blockstate file". Items, including block-placing items, have separate models.
+
+Let's give the border block a texture with white background and black borders, like this. Most block textures are 16-by-16 square. This is placed at `/textures/block/border.png`, inside the mod resource pack (`/assets/(mod id)/`).
+
+![](/reference/latest/src/main/resources/assets/fabric-docs-reference/textures/block/border.png)
+
+This texture is then referenced by a model file named `border.json`, placed under `/models/block` directory inside the mod resource pack. Since the block has the same texture on all six sides, we use `cube_all` as the parent model.
+
+@[code lang=json](@/reference/latest/src/main/resources/assets/fabric-docs-reference/models/block/border.json)
```

Comment:
This is good so dont change anything, but I do think we want to encorage the use of data generation, especially as its likely going to be a lot more important in future updates.